api/XDS_PROTOCOL.md

```diff
@@ -345,6 +345,11 @@ resources to avoid resending them over the network.
 
 ![Incremental reconnect example](diagrams/incremental-reconnect.svg)
 
+#### Resource names
+Resources are identified by a resource name or an alias. Aliases of a resource, if present, can be identified by the alias field in the resource of a DeltaDiscoveryResponse. The resource names
+will be returned in either the name field in the resource of a DeltaDiscoveryResponse or in the actual body
+of the resource. Either the alias or the resource name can be passed in the resource_names_subscribe and resource_names_unsubscribe fields of a DeltaDiscoveryRequest to identify a specific resource. 
```

Comment:
Here's a scenario that I'm wondering about:
1. Envoy has listener X, it subscribes to VirtualHost Foo via alias M.
2. Envoy later has listener Y that wants Foo, but subscribes via alias N to Foo.
3. Listener X is removed, and Envoy consequently unsubscribes from M.
At this point, the management server and Envoy have decided to remove Foo.

Is the solution to have Envoy make use of the `aliases` field and internally reference count its use of a given resource, only unsubscribing when its internal use has gone down to zero? If so, maybe we could spell this out more concretely here, since it's implying some client behavior that is part of the xDS contract.

Reply:
Done. Thanks.